[Inference Providers] Update partners API documentation (#1717)
huggingface/hub-docs: docs/inference-providers/register-as-a-provider.md

- `hfModel` is the model id on the Hub's side.
- `providerModel` is the model id on your side (can be the same or different).

> The output of such route is a mapping ID that you can use later to update the mapping's status; or to delete it.

Reviewer: (nit)
```suggestion
The output of this route is a mapping ID that you can later use to update the mapping's status or delete it.
```

### Using a tag-filter to map several HF models to a single inference endpoint

> We also support mapping HF models based on their `tags`.
> 
> This is useful to, for example, automatically map LoRA adapters to a single Inference Endpoint on your side.

Reviewer: (nit)
```suggestion
We also support mapping HF models based on their `tags`. Using tag filters, you can automatically map multiple HF models to a single inference endpoint on your side.
For example, any model tagged with both `lora` and `base_model:adapter:black-forest-labs/FLUX.1-dev` can be mapped to your Flux-dev LoRA inference endpoint.

```

<Tip>

> Important note: the client library (Javascript) must be able to handle LoRA weights for your provider. Check out [fal's implementation](https://github.com/huggingface/huggingface.js/blob/904964c9f8cd10ed67114ccb88b9028e89fd6cad/packages/inference/src/providers/fal-ai.ts#L78-L124) for more details. 

Reviewer: ```suggestion
Important: Make sure that the JS client library can handle LoRA weights for your provider. Check out [fal's implementation](https://github.com/huggingface/huggingface.js/blob/904964c9f8cd10ed67114ccb88b9028e89fd6cad/packages/inference/src/providers/fal-ai.ts#L78-L124) for more details. 
```

</Tip>

The API is as follows:

```http
POST /api/partners/{provider}/models
```
Create a new mapping item, with the following body (JSON-encoded):

```json
{
    "type": "tag-filter", // required
    "task": "WidgetType", // required
    "tags": ["string"], // required: any HF model with all of those tags will be mapped to providerModel
    "providerModel": "string", // required: the partner's "model id" i.e. id on your side
    "adapterType": "lora", // required: only "lora" is supported at the moment
    "status": "live" | "staging" // Optional: defaults to "staging". "staging" models are only available to members of the partner's org, then you switch them to "live" when they're ready to go live
}
```

- `task`, also known as `pipeline_tag` in the HF ecosystem, is the type of model / type of API
(examples: "text-to-image", "text-generation", but you should use "conversational" for chat models)
- `tags` is the set of model tags to match. For example, to match all LoRAs of Flux, you can use: `["lora", "base_model:adapter:black-forest-labs/FLUX.1-dev"]` 
- `providerModel` is the model id on your side (can be the same or different).

> - `adapterType` is a literal value designed to help client libraries interpret how to request your API. The only supported value at the moment is `"lora"`.

Reviewer: ```suggestion
- `adapterType` is a literal value designed that helps client libraries interpret how to call your API. The only supported value at the moment is `"lora"`.
```

> The output of such route is a mapping ID that you can use later to update the mapping's status; or to delete it.

Reviewer: ```suggestion
The output of this route is a mapping ID that you can later use to update the mapping's status or delete it.
```
